runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -69,15 +71,16 @@
 	private static final String CONFIG_GROUP = "groundMarker";
 	private static final String MARK = "Mark tile";
 	private static final String WALK_HERE = "Walk here";
+	private static final String REGION_PREFIX = "region_";
 
-	private static final Gson gson = new Gson();
+	private static final Gson GSON = new Gson();
 
 	@Getter(AccessLevel.PACKAGE)
 	@Setter(AccessLevel.PACKAGE)
 	private boolean hotKeyPressed;
 
 	@Getter(AccessLevel.PACKAGE)
-	private final List<WorldPoint> points = new ArrayList<>();
+	private final Map<GroundMarkerPoint, Collection<WorldPoint>> points = new HashMap<>();
```

Comment:
This should probably instead just be a list of some private class which contains scene x/y/z and Color

Reply:
At that point, it seems easier to add serialization for worldpoints to configmanager and use a worldpoint directly since that removes the need to handle scene logic within the plugin. At this point, I'm inclined to rewrite GroundMarkerPoint as a wrapper for a worldpoint and color and go from there.

Reply:
I think it would be better to use a separate class.

Reply:
I think we're talking about the same thing. Let me know if what I'm doing now is different than what you had in mind.